browser/brave_news/brave_news_tab_helper.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_BRAVE_NEWS_BRAVE_NEWS_TAB_HELPER_H_
+#define BRAVE_BROWSER_BRAVE_NEWS_BRAVE_NEWS_TAB_HELPER_H_
+
+#include <string>
+#include <vector>
+
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "base/scoped_observation.h"
+#include "brave/components/brave_today/browser/brave_news_controller.h"
+#include "brave/components/brave_today/browser/publishers_controller.h"
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+class Page;
```

Comment:
Interesting, why not? Is it because there's an include/forward ref in the base class `WebContentsObserver` so the symbol is already there when I override `PrimaryPageChanged(content::Page& page)`

I think this should've been
```suggestion
namespace content {
class Page;
}
```

Should I remove the forward declaration completely (I've just moved it into the content namespace), but obviously the code seems to work without the forward declaration (or with the wrong one :laughing: )

Reply:
because this type is already used in the method you're *overriding* means that it's 100% declared for you. don't need to add this forward at all.